howso/residuals.amlg

```diff
@@ -1308,12 +1446,56 @@
 										;compute from the existing feature max-min value or bounds if provided
 										(max
 											(-
-												(get !featureMarginalStatsMap (list weight_feature feature "max"))
-												(get !featureMarginalStatsMap (list weight_feature feature "min"))
+												(if (size feature_grouping)
+													(apply "max"
+														(filter
+															(lambda (!= (null) (current_value)))
```

Comment:
(filter) by default only leaves non-null values, therefore there's no need for these (lambda) that match on nun-null, you can just do `(filter (map ... `